ush/python/pyobsforge/obsdb/obsdb.py

```diff
@@ -55,26 +55,30 @@ def execute_query(self, query: str, params: tuple = None) -> list:
         return results
 
     def get_valid_files(self,
-                        da_cycle: str,
-                        window_hours: int = 3,
+                        window_begin: datetime,
+                        window_end: datetime,
                         instrument: str = None,
                         satellite: str = None,
                         obs_type: str = None,
-                        cutoff_delta: int = 0) -> list:
+                        check_receipt: str = "none") -> list:
         """
-        Retrieve a list of observation files within a DA window, possibly filtered by instrument,
-        satellite, observation type, and cutoff delta (known latency to emulate the early cycle if needed).
+        Retrieve a list of observation files within a specified time window, possibly filtered by instrument,
+        satellite, and observation type. The check_receipt parameter can be 'gdas', 'gfs', or 'none'.
+
+        :param window_begin: Start of the time window (datetime object).
+        :param window_end: End of the time window (datetime object).
+        :param instrument: (Optional) Filter by instrument name.
+        :param satellite: (Optional) Filter by satellite name.
+        :param obs_type: (Optional) Filter by observation type.
+        :param check_receipt: (Optional) Specify receipt time check ('gdas', 'gfs', or 'none').
```

Comment:
This is fine for now, but we shouldn't use those names, because this should be a flexible enough tool to work for hourly cycling or regional models, etc.